Tests don't compile on Windows

Trying to run tests on Windows fails with following compilation errors:

```
error[E0425]: cannot find value `MAP_ANONYMOUS` in module `libc`
   --> src\simd.rs:668:48
    |
668 |     const MAP_ANONYMOUS: libc::int32_t = libc::MAP_ANONYMOUS;
    |                                                ^^^^^^^^^^^^^ not found in `libc`
help: possible candidate is found in another module, you can import it into scope
    |
298 |     use simd::test::MAP_ANONYMOUS;
    |

error[E0425]: cannot find value `PROT_READ` in module `libc`
   --> src\simd.rs:684:30
    |
684 |             let prot = libc::PROT_READ | libc::PROT_WRITE;
    |                              ^^^^^^^^^ not found in `libc`

error[E0425]: cannot find value `PROT_WRITE` in module `libc`
   --> src\simd.rs:684:48
    |
684 |             let prot = libc::PROT_READ | libc::PROT_WRITE;
    |                                                ^^^^^^^^^^ not found in `libc`

error[E0425]: cannot find value `MAP_PRIVATE` in module `libc`
   --> src\simd.rs:685:31
    |
685 |             let flags = libc::MAP_PRIVATE | MAP_ANONYMOUS;
    |                               ^^^^^^^^^^^ not found in `libc`

error[E0425]: cannot find function `mmap` in module `libc`
   --> src\simd.rs:689:36
    |
689 |             let first_page = libc::mmap(addr, length, prot, flags, fd, offset);
    |                                    ^^^^ not found in `libc`

error[E0425]: cannot find value `PROT_NONE` in module `libc`
   --> src\simd.rs:699:34
    |
699 |                 let prot = libc::PROT_NONE;
    |                                  ^^^^^^^^^ not found in `libc`

error[E0425]: cannot find function `mprotect` in module `libc`
   --> src\simd.rs:701:45
    |
701 |                 let mprotect_retval = libc::mprotect(addr, length, prot);
    |                                             ^^^^^^^^ not found in `libc`

error: aborting due to 7 previous errors

For more information about this error, try `rustc --explain E0425`.
error: Could not compile `jetscii`.

To learn more, run the command again with --verbose.
```

It would be better to use platform-independent crate like https://crates.io/crates/region.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -18,5 +18,6 @@ pattern = []
 
 [dev-dependencies]
 proptest = "0.8.3"
-libc = "0.2"
 lazy_static = "1.0.0"
+region = "1.0.0"
+memmap = "0.7.0"

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -145,10 +145,12 @@
 #[macro_use]
 extern crate lazy_static;
 #[cfg(test)]
-extern crate libc;
+extern crate memmap;
 #[cfg(test)]
 #[macro_use]
 extern crate proptest;
+#[cfg(test)]
+extern crate region;
 
 use std::marker::PhantomData;
 

--- a/src/simd.rs
+++ b/src/simd.rs
@@ -295,9 +295,10 @@ impl<'a, 'b> PackedCompareControl for &'b ByteSubstring<'a> {
 
 #[cfg(test)]
 mod test {
-    use libc;
     use proptest::prelude::*;
-    use std::{fmt, ptr, str};
+    use std::{fmt, str};
+    use memmap::MmapMut;
+    use region::Protection;
 
     use super::*;
 
@@ -661,52 +662,28 @@ mod test {
         }
     }
 
-    #[cfg(target_os = "macos")]
-    const MAP_ANONYMOUS: libc::int32_t = libc::MAP_ANON;
-
-    #[cfg(not(target_os = "macos"))]
-    const MAP_ANONYMOUS: libc::int32_t = libc::MAP_ANONYMOUS;
-
-    fn alloc_guarded_string(value: &str, protect: bool) -> &'static str {
+    fn with_guarded_string(value: &str, f: impl FnOnce(&str)) {
         // Allocate a string that ends directly before a
         // read-protected page.
-        //
-        // This function leaks two pages of memory per call, which is
-        // acceptable for use in tests only.
 
-        const PAGE_SIZE: usize = 4096;
-        assert!(value.len() <= PAGE_SIZE);
+        let page_size = region::page::size();
+        assert!(value.len() <= page_size);
 
+        // Map two rw-accessible pages of anonymous memory
+        let mut mmap = MmapMut::map_anon(2 * page_size).unwrap();
+
+        let (first_page, second_page) = mmap.split_at_mut(page_size);
+
+        // Prohibit any access to the second page, so that any attempt
+        // to read or write it would trigger a segfault
         unsafe {
-            // Map two rw-accessible pages of anonymous memory
-            let addr = 0 as *mut libc::c_void;
-            let length = 2 * PAGE_SIZE as libc::size_t;
-            let prot = libc::PROT_READ | libc::PROT_WRITE;
-            let flags = libc::MAP_PRIVATE | MAP_ANONYMOUS;
-            let fd = -1;
-            let offset = 0;
-
-            let first_page = libc::mmap(addr, length, prot, flags, fd, offset);
-            assert!(!first_page.is_null());
-
-            let second_page = first_page.offset(PAGE_SIZE as isize);
-
-            if protect {
-                // Prohibit any access to the second page, so that any attempt
-                // to read or write it would trigger a segfault
-                let addr = second_page;
-                let length = PAGE_SIZE as libc::size_t;
-                let prot = libc::PROT_NONE;
-
-                let mprotect_retval = libc::mprotect(addr, length, prot);
-                assert_eq!(0, mprotect_retval);
-            }
-
-            // Copy bytes to the end of the first page
-            let start = second_page.offset(-(value.len() as isize)) as *mut u8;
-            ptr::copy_nonoverlapping(value.as_ptr(), start, value.len());
-            str::from_utf8_unchecked(slice::from_raw_parts(start, value.len()))
+            region::protect(second_page.as_ptr(), page_size, Protection::None).unwrap();
         }
+
+        // Copy bytes to the end of the first page
+        let dest = &mut first_page[page_size - value.len()..];
+        dest.copy_from_slice(value.as_bytes());
+        f(unsafe { str::from_utf8_unchecked(dest) });
     }
 
     #[test]
@@ -721,17 +698,17 @@ mod test {
 
         // Allocate a 16-byte string at page boundary.  To verify this
         // test, set protect=false to prevent segfaults.
-        let text = alloc_guarded_string("0123456789abcdef", true);
+        with_guarded_string("0123456789abcdef", |text| {
+            // Will search for the last char
+            let needle = simd_bytes!(b'f');
 
-        // Will search for the last char
-        let needle = simd_bytes!(b'f');
-
-        // Check all suffixes of our 16-byte string
-        for offset in 0..text.len() {
-            let tail = &text[offset..];
-            unsafe {
-                assert_eq!(Some(tail.len() - 1), needle.find(tail.as_bytes()));
+            // Check all suffixes of our 16-byte string
+            for offset in 0..text.len() {
+                let tail = &text[offset..];
+                unsafe {
+                    assert_eq!(Some(tail.len() - 1), needle.find(tail.as_bytes()));
+                }
             }
-        }
+        });
     }
 }

